shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java

```diff
@@ -263,26 +341,153 @@ private static int clampIndexToEditable(int index, Editable editable) {
     return clamped;
   }
 
+  public int getOffsetBefore(int offset) {
+    if (offset <= 1) {
```

Comment:
I would also provide a link to Android's implementation as reference for an existing version of this logic. https://android.googlesource.com/platform/frameworks/base/+/refs/heads/android10-s3-release/core/java/android/text/method/BaseKeyListener.java#111 Seems like a good version to link to. Master may change so this is the android 10 version.

Reply:
This is also in Chromium's Blink: https://source.chromium.org/chromium/chromium/src/+/84.0.4131.5:third_party/blink/renderer/core/editing/state_machines/backspace_state_machine.cc

Reply:
My refrence for handling this behavior was the android source code, too.